framework/src/mesh/MooseMesh.C

```diff
@@ -3313,7 +3313,14 @@ MooseMesh::buildFiniteVolumeInfo() const
 {
   if (!_finite_volume_info_dirty)
     return;
-  _finite_volume_info_dirty = false;
+
+  // Only one thread must build the finite volume info at a time
+  // but no need to lock if we are not in a threaded region
+  std::unique_ptr<Threads::spin_mutex::scoped_lock> lock;
+  if (Threads::in_threads)
+    lock = std::make_unique<Threads::spin_mutex::scoped_lock>(Threads::spin_mtx);
```

Comment:
is there a cleaner way to do this?

The lock needs to stay in scope and I dont want to be creating it if we re not threaded (which we usually are not, but in my use here we are) because it's too slow (fails debug recipe pretty hard)

Reply:
We just shouldn't be calling this method in a threaded region. That's the fix: ensuring that the face information is already built before calling e.g. `faceInfo()` in a threaded region. Then there will be no need for locks